@i3arnon seems you got some build errors:

```
/__w/1/s/src/coreclr/vm/ceeload.cpp:12445:37: error: out-of-line definition of 'Create' does not match any declaration in 'ReflectionModule'
ReflectionModule *ReflectionModule::Create(Assembly *pAssembly, PEFile *pFile, AllocMemTracker *pamTracker, LPCWSTR szName, BOOL fIsTransient)
                                    ^~~~~~
/__w/1/s/src/coreclr/vm/ceeload.cpp:12472:14: error: no member named 'SetIsTransient' in 'ReflectionModule'
    pModule->SetIsTransient(fIsTransient ? true : false);
    ~~~~~~~  ^
/__w/1/s/src/coreclr/vm/ceeload.cpp:12501:5: error: use of undeclared identifier 'm_fIsTransient'
    m_fIsTransient = false;
```

--- a/src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.cs
+++ b/src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.cs
@@ -305,10 +305,6 @@ namespace System.Reflection.Emit
         public ModuleBuilder DefineDynamicModule(string name) =>
             DefineDynamicModuleInternal(name, emitSymbolInfo: false);
 
-        [DynamicSecurityMethod] // Methods containing StackCrawlMark local var has to be marked DynamicSecurityMethod.
-        public ModuleBuilder DefineDynamicModule(string name, bool emitSymbolInfo) =>
-            DefineDynamicModuleInternal(name, emitSymbolInfo);
-
         private ModuleBuilder DefineDynamicModuleInternal(string name, bool emitSymbolInfo)
         {
             lock (SyncRoot)

--- a/src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/ModuleBuilder.cs
+++ b/src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/ModuleBuilder.cs
@@ -1598,20 +1598,6 @@ namespace System.Reflection.Emit
         // Regardless, this is a reliability bug.
         internal ISymbolWriter? GetSymWriter() => _iSymWriter;
 
-        public ISymbolDocumentWriter? DefineDocument(string url, Guid language, Guid languageVendor, Guid documentType)
-        {
-            // url cannot be null but can be an empty string
-            if (url == null)
-            {
-                throw new ArgumentNullException(nameof(url));
-            }
-
-            lock (SyncRoot)
-            {
-                return DefineDocumentNoLock(url, language, languageVendor, documentType);
-            }
-        }
-
         private ISymbolDocumentWriter? DefineDocumentNoLock(string url, Guid language, Guid languageVendor, Guid documentType)
         {
             if (_iSymWriter == null)

--- a/src/coreclr/vm/ceeload.cpp
+++ b/src/coreclr/vm/ceeload.cpp
@@ -12442,7 +12442,7 @@ idMethodSpec Module::LogInstantiatedMethod(const MethodDesc * md, ULONG flagNum)
 // ===========================================================================
 
 /* static */
-ReflectionModule *ReflectionModule::Create(Assembly *pAssembly, PEFile *pFile, AllocMemTracker *pamTracker, LPCWSTR szName, BOOL fIsTransient)
+ReflectionModule *ReflectionModule::Create(Assembly *pAssembly, PEFile *pFile, AllocMemTracker *pamTracker, LPCWSTR szName)
 {
     CONTRACT(ReflectionModule *)
     {
@@ -12467,9 +12467,6 @@ ReflectionModule *ReflectionModule::Create(Assembly *pAssembly, PEFile *pFile, A
     ReflectionModuleHolder pModule(new (pMemory) ReflectionModule(pAssembly, token, pFile));
 
     pModule->DoInit(pamTracker, szName);
-
-    // Set this at module creation time. The m_fIsTransient field should never change during the lifetime of this ReflectionModule.
-    pModule->SetIsTransient(fIsTransient ? true : false);
 
     RETURN pModule.Extract();
 }
@@ -12498,7 +12495,6 @@ ReflectionModule::ReflectionModule(Assembly *pAssembly, mdFile token, PEFile *pF
     m_pCeeFileGen = NULL;
     m_pDynamicMetadata = NULL;
     m_fSuppressMetadataCapture = false;
-    m_fIsTransient = false;
 }
 
 HRESULT STDMETHODCALLTYPE CreateICeeGen(REFIID riid, void **pCeeGen);

--- a/src/coreclr/vm/commodule.cpp
+++ b/src/coreclr/vm/commodule.cpp
@@ -709,28 +709,6 @@ void QCALLTYPE COMModule::SetModuleName(QCall::ModuleHandle pModule, LPCWSTR wsz
 // Return a type spec token given a byte array
 //
 //******************************************************************************
-BOOL QCALLTYPE COMModule::IsTransient(QCall::ModuleHandle pModule)
-{
-    QCALL_CONTRACT;
-
-    BOOL fIsTransient = FALSE;
-
-    BEGIN_QCALL;
-
-    /* Only reflection modules can be transient */
-    if (pModule->IsReflection())
-        fIsTransient = pModule->GetReflectionModule()->IsTransient();
-
-    END_QCALL;
-
-    return fIsTransient;
-}
-
-//******************************************************************************
-//
-// Return a type spec token given a byte array
-//
-//******************************************************************************
 mdTypeSpec QCALLTYPE COMModule::GetTokenFromTypeSpec(QCall::ModuleHandle pModule, LPCBYTE pSignature, INT32 sigLength)
 {
     QCALL_CONTRACT;

--- a/src/coreclr/vm/ecalllist.h
+++ b/src/coreclr/vm/ecalllist.h
@@ -324,7 +324,6 @@ FCFuncStart(gCOMModuleFuncs)
     QCFuncElement("GetScopeName", COMModule::GetScopeName)
     FCFuncElement("GetTypes", COMModule::GetTypes)
     QCFuncElement("GetFullyQualifiedName", COMModule::GetFullyQualifiedName)
-    QCFuncElement("nIsTransientInternal", COMModule::IsTransient)
     FCFuncElement("IsResource", COMModule::IsResource)
 FCFuncEnd()
 

